apis/peripherals/adc/src/lib.rs

```diff
@@ -22,10 +22,22 @@ impl<S: Syscalls> Adc<S> {
             .to_result::<u32, ErrorCode>()
             .and(Ok(()))
     }
+    //Returns the number of channels
+    pub fn get_number_of_channels() -> Result<usize, ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0)
+            .to_result::<u32, ErrorCode>()
+            .and_then(|number_channels| number_channels.try_into().map_err(|_| ErrorCode::Fail))
+    }
 
     // Initiate a sample reading
-    pub fn read_single_sample() -> Result<(), ErrorCode> {
-        S::command(DRIVER_NUM, SINGLE_SAMPLE, 0, 0).to_result()
+    pub fn read_single_sample(channel: usize) -> Result<(), ErrorCode> {
+        S::command(
+            DRIVER_NUM,
+            SINGLE_SAMPLE,
+            (channel as usize).try_into().unwrap(),
```

Comment:
```suggestion
            channel as u32,
```

I would avoid the panic as most probably `usize` will be 32 bits.

@jrvanwhy I would like your input on this.

Reply:
Without looking at the entire CL: it looks like `read_single_sample` should take `channel` as a `u32`.

One question to ask that helps decide `usize` versus `u32` is "in the unit test environment, which runs on 64-bit platforms, should this number get larger?" I don't think a unit test environment should support more channels than the real hardware, so this should be a `u32`. `usize` makes sense for memory addresses and offsets, but not necessarily for other hardware addresses.